source/extensions/common/wasm/wavm/wavm.cc

```diff
@@ -178,36 +179,37 @@ template <typename F> EnvoyHandlerBase* MakeEnvoyHandler(F handler) {
 class Wavm : public WasmVm {
 public:
   Wavm() = default;
+  ;
   ~Wavm() override;
 
   // WasmVm
   absl::string_view vm() override { return WasmVmNames::get().Wavm; }
   bool clonable() override { return true; };
   std::unique_ptr<WasmVm> clone() override;
-  bool initialize(const std::string& code, absl::string_view name, bool allow_precompiled) override;
+  bool load(const std::string& code, bool allow_precompiled) override;
+  void link(absl::string_view debug_name) override;
   void start(Context* context) override;
   void* allocMemory(uint32_t size, uint32_t* pointer) override;
   absl::string_view getMemory(uint32_t pointer, uint32_t size) override;
   bool setMemory(uint32_t pointer, uint32_t size, void* data) override;
-
-  WAVM::Runtime::Memory* memory() { return memory_; }
-  WAVM::Runtime::Context* context() { return context_; }
-  WAVM::Runtime::ModuleInstance* moduleInstance() { return moduleInstance_; }
-  WAVM::Runtime::ModuleInstance* envoyModuleInstance() { return moduleInstance_; }
+  void makeModule(absl::string_view name) override;
+  absl::string_view getUserSection(absl::string_view name, bool* present) override;
 
   void GetFunctions();
   void RegisterCallbacks();
 
   bool hasInstantiatedModule_ = false;
```

Comment:
IMHO all the belows should be private...

Reply:
This class is entirely within this file so it is well encapsulated already and the purpose of 'private' is to provide proper encapsulation which has already been achieved.  I'll change it to a struct.

Reply:
Thanks for the explanation. I should add nit prefix for this comment